backend/server.py

```
@@ -1512,29 +1512,42 @@ async def lifespan(app_: "FastAPI"):
+    # deduplication. The cleanup is idempotent + best-effort — if the process
+    # exits before it finishes, the next cold start re-runs it, and the
+    # per-minute Cloudflare Cron /api/scheduler/tick also runs force_cleanup
+    # on every tick. Don't block the first request on cleanup that can happen
+    # 60s later with no user-visible impact.
+    async def _deferred_startup_cleanup() -> None:
+        try:
+            cleanup = await SCHEDULER.force_cleanup()
+            if cleanup.get("deleted", 0) > 0 or cleanup.get("deduped", 0) > 0:
+                log.info("Startup schedule cleanup: deleted=%d deduped=%d total=%d",
+                         cleanup["deleted"], cleanup["deduped"], cleanup["total"])
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)